src/librustc_typeck/astconv.rs

```diff
@@ -2358,10 +2358,9 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                     break;
                 }
             }
-            for binding in segment.generic_args().bindings {
+            if let [binding, ..] = segment.generic_args().bindings {
```

Comment:
This is a bit out of cache for me, but I think the reason we broke out of the loop immediately was because we only wanted to emit the first error in this situation, to avoid overloading the user with error messages. We should add a comment here to that effect, because it's not clear that's what's happening at the moment.

Reply:
Thanks, I added a comment.